### PR TITLE
iOS-3251 Version history | Refactor preview flag + add version mode

### PR DESCRIPTION
**Pull request by joe-pusya**
https://github.com/user-attachments/assets/aa350cfe-62cf-48de-b42e-a62eceedc6c5

### CODE REVIEW

File: Anytype/Sources/Models/Documents/Document/BaseDocument.swift [hunk 1]
```diff
@@ -102,30 +105,31 @@ final class BaseDocument: BaseDocumentProtocol {
...
     }
     
     @MainActor
-    func openForPreview() async throws {
-        guard forPreview else {
-            anytypeAssertionFailure("Document created for handling. You should use open() method.")
-            return
+    func update() async throws {
+        switch mode {
+        case .handling, .preview:
```
Comment (mgolovko) [inline Anytype/Sources/Models/Documents/Document/BaseDocument.swift:123]: update available for handling?

File: Anytype/Sources/PresentationLayer/TextEditor/EditorPage/Views/Settings/History/Object/ObjectVersionViewModel.swift [hunk 2]
```diff
@@ -17,18 +18,20 @@ final class ObjectVersionViewModel: ObservableObject {
...
         self.data = data
-        self.screenData = currentScreenData()
     }
     
-    private func currentScreenData() -> EditorScreenData? {
+    func setupObject() async {
+        self.screenData = await currentScreenData()
+    }
+    
+    private func currentScreenData() async -> EditorScreenData? {
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/TextEditor/EditorPage/Views/Settings/History/Object/ObjectVersionViewModel.swift:29]: why async?

**Review by mgolovko — APPROVED** [review]

File: Anytype/Sources/PresentationLayer/TextEditor/EditorPage/Views/Settings/History/Object/ObjectVersionViewModel.swift (hunk 2 above)
Comment (joe-pusya) [reply Anytype/Sources/PresentationLayer/TextEditor/EditorPage/Views/Settings/History/Object/ObjectVersionViewModel.swift:29]: Not needed, forgot to remove, fixed in the following PR

File: Anytype/Sources/Models/Documents/Document/BaseDocument.swift (hunk 1 above)
Comment (joe-pusya) [reply Anytype/Sources/Models/Documents/Document/BaseDocument.swift]: added assert for this case 🙏

**Pull request merged**